qiskit/algorithms/minimum_eigensolvers/sampling_vqe.py

```diff
@@ -0,0 +1,301 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Variational Quantum Eigensolver algorithm, optimized for diagonal Hamiltonians."""
+
+from __future__ import annotations
+
+import logging
+from time import time
+from collections.abc import Callable, Sequence
+
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseSampler
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from ..exceptions import AlgorithmError
+from ..list_or_dict import ListOrDict
+from ..optimizers import Minimizer, Optimizer
+from ..variational_algorithm import VariationalAlgorithm, VariationalResult
+from .sampling_mes import (
+    SamplingMinimumEigensolver,
+    SamplingMinimumEigensolverResult,
+)
+
+from .diagonal_estimator import _DiagonalEstimator
+
+logger = logging.getLogger(__name__)
+
+
+class SamplingVQE(VariationalAlgorithm, SamplingMinimumEigensolver):
+    r"""The Variational Quantum Eigensolver algorithm.
```

Comment:
Maybe we need to qualify this text here too by adding something like `, optimized for diagonal Hamiltonians` as is done in the class docstring above